learning_resources_search/api.py

```diff
@@ -1,23 +1,41 @@
 """API for general search-related functionality"""
 import re
+from base64 import urlsafe_b64encode
 
 from opensearch_dsl import Search
 
 from learning_resources_search.connection import get_default_alias_name
 from learning_resources_search.constants import (
+    CONTENT_FILE_TYPE,
+    COURSE_TYPE,
+    DEPARTMENT_QUERY_FIELDS,
     LEARNING_RESOURCE_QUERY_FIELDS,
     LEARNING_RESOURCE_SEARCH_FILTERS,
     LEARNING_RESOURCE_TYPES,
+    RESOURCEFILE_QUERY_FIELDS,
     RUN_INSTRUCTORS_QUERY_FIELDS,
     RUNS_QUERY_FIELDS,
     SEARCH_NESTED_FILTERS,
     TOPICS_QUERY_FIELDS,
 )
 
-SIMILAR_RESOURCE_RELEVANT_FIELDS = ["title", "short_description"]
 LEARN_SUGGEST_FIELDS = ["title.trigram", "description.trigram"]
 
 
+def gen_content_file_id(key):
+    """
+    Generate the OpenSearch document id for a ContentFile
+
+    Args:
+        key (str): The key of a ContentFile object
+
+    Returns:
+        str: The OpenSearch document id for this object
+    """
+    safe_key = urlsafe_b64encode(key.encode("utf-8")).decode("utf-8").rstrip("=")
```

Comment:
Noticed some unique constraint errors when running `backpopulate_mitxonline_files` (had also run `backpopulate_xpro_files` previously but wasn't paying attention during that one).  Not sure why, didn't investigate further yet, but could be that different courses have contentfiles with the same key.  Maybe can use the `ContentFile.id` value instead?